Documentation/gettingstarted/k8s-install-default.rst

```diff
@@ -214,6 +203,25 @@ You can install Cilium on any Kubernetes cluster. Pick one of the options below:
 
            cilium install --azure-resource-group "${AZURE_RESOURCE_GROUP}"
 
+       Now that Cilium is installed we can remove the taint on the new node pool and then delete the 
+       original node pool. This will allow for the Pods on the original system node pool to be rescheduled on
+       the newly created one:
```

Comment:
Maybe this is not clear from the instructions but this should not be necessary. Once Cilium is ready, it will remove the node-taints from the nodes that it is deployed on. The user does not need to remove the taints.

Reply:
That's interesting. On a new cluster with Cilium `v1.10.2`, (installed using the cilium cli on v0.8.5) the taint is still in the node pool  even after cilium is installed. If I don't remove the taint than the original node pool doesn't get removed and the pods there can't be scheduled.

```bash
$ kubectl get nodes -o jsonpath='{range .items[]}{.metadata.name}{"\t"}{.spec.taints[].effect}{"\n"}{end}'
aks-nodepool1-36482404-vmss000000	NoSchedule
```

```bash
 $ cilium version 
cilium-cli: v0.8.5 compiled with go1.16.5 on linux/amd64
cilium image (default): v1.10.2
cilium image (stable): v1.10.3
```

This is a view of the pods on all namespaces:

```bash
$ kubectl get po -A
NAMESPACE     NAME                                  READY   STATUS    RESTARTS   AGE
kube-system   azure-cni-networkmonitor-2kgz2        1/1     Running   0          12m
kube-system   azure-cni-networkmonitor-lt8vs        1/1     Running   0          12m
kube-system   azure-cni-networkmonitor-mpc62        1/1     Running   0          8m30s
kube-system   azure-cni-networkmonitor-rqpvc        1/1     Running   0          12m
kube-system   azure-cni-networkmonitor-x6qxx        1/1     Running   0          8m31s
kube-system   azure-ip-masq-agent-9vfff             1/1     Running   0          12m
kube-system   azure-ip-masq-agent-jc94z             1/1     Running   0          8m31s
kube-system   azure-ip-masq-agent-jlm85             1/1     Running   0          12m
kube-system   azure-ip-masq-agent-l8hgn             1/1     Running   0          12m
kube-system   azure-ip-masq-agent-qbb5t             1/1     Running   0          8m30s
kube-system   cilium-2lfzt                          1/1     Running   0          7m50s
kube-system   cilium-7gmc8                          1/1     Running   0          7m50s
kube-system   cilium-82lrz                          1/1     Running   0          7m50s
kube-system   cilium-lwbxm                          1/1     Running   0          7m50s
kube-system   cilium-operator-5dbf4f7947-5pcs2      1/1     Running   0          7m50s
kube-system   cilium-sfnpm                          1/1     Running   0          7m50s
kube-system   coredns-9d6c6c99b-twt4w               1/1     Running   0          7m9s
kube-system   coredns-9d6c6c99b-zw7n9               0/1     Pending   0          5m45s
kube-system   coredns-autoscaler-599949fd86-gbk8s   0/1     Pending   0          5m45s
kube-system   konnectivity-agent-5697c559d4-95tfc   0/1     Pending   0          5m45s
kube-system   konnectivity-agent-5697c559d4-pqkpk   1/1     Running   0          13m
kube-system   kube-proxy-84cfq                      1/1     Running   0          12m
kube-system   kube-proxy-b4fqn                      1/1     Running   0          8m31s
kube-system   kube-proxy-hvcbv                      1/1     Running   0          12m
kube-system   kube-proxy-vclvl                      1/1     Running   0          12m
kube-system   kube-proxy-znnc4                      1/1     Running   0          8m29s
kube-system   metrics-server-77c8679d7d-fqh8c       0/1     Pending   0          5m45s
```

and the details on the cilium install:

```bash
🔮 Auto-detected Kubernetes kind: AKS
✨ Running "AKS" validation checks
✅ Detected az binary
ℹ️  using Cilium version "v1.10.2"
🔮 Auto-detected cluster name: dcasati-23222
🔮 Auto-detected IPAM mode: azure
🔮 Auto-detected datapath mode: azure
🔮 Custom datapath mode: azure
✅ Derived Azure subscription ID b4b07aea-0ece-XXX from subscription Diego Casati 
✅ Derived Azure AKS node resource group MC_rg-aks-cilium-test_dcasati-23222_westus2 from resource group rg-aks-cilium-test
🚀 Creating Azure Service Principal for Cilium operator...
✅ Created Azure Service Principal for Cilium operator with App ID 63176db1-XXX and Tenant ID 72f988bf-86f1-XXX
ℹ️ Its RBAC privileges are restricted to the AKS node resource group MC_rg-aks-cilium-test_dcasati-23222_westus2
🔑 Generating CA...
2021/08/10 11:45:56 [INFO] generate received request
2021/08/10 11:45:56 [INFO] received CSR
2021/08/10 11:45:56 [INFO] generating key: ecdsa-256
2021/08/10 11:45:56 [INFO] encoded CSR
2021/08/10 11:45:57 [INFO] signed certificate with serial number 278832086850127796423093726388868664644170313118
🔑 Generating certificates for Hubble...
2021/08/10 11:45:57 [INFO] generate received request
2021/08/10 11:45:57 [INFO] received CSR
2021/08/10 11:45:57 [INFO] generating key: ecdsa-256
2021/08/10 11:45:57 [INFO] encoded CSR
2021/08/10 11:45:57 [INFO] signed certificate with serial number 256752994521428528557059395585988332722703545244
🚀 Creating Service accounts...
🚀 Creating Cluster roles...
🚀 Creating ConfigMap...
🚀 Creating Agent DaemonSet...
🚀 Creating Operator Deployment...
⌛ Waiting for Cilium to be installed...
⌛ Waiting for Cilium to become ready before restarting unmanaged pods...
♻️  Restarting unmanaged pods...
♻️  Restarted unmanaged pod kube-system/coredns-autoscaler-599949fd86-b6m52
♻️  Restarted unmanaged pod kube-system/metrics-server-77c8679d7d-d7xbm
✅ Cilium was successfully installed! Run 'cilium status' to view installation health
The behavior of this command has been altered by the following extension: aks-preview
(DeleteVMSSAgentPoolFailed) Category: ClientError; SubCode: K8SFailedDrainNode; Dependency: Kubernetes API; OrginalError: Node 'aks-nodepool1-36482404-vmss000002' failed to be drained with error: 'Category: InternalError; SubCode: DrainDidNotComplete; Dependency: Kubernetes API; OrginalError: Drain did not complete pods [konnectivity-agent-5697c559d4-pqkpk]; AKSTeam: NodeProvisioning'. Check Pod Disruption BudgetsNode 'aks-nodepool1-36482404-vmss000000' failed to be drained with error: 'Category: InternalError; SubCode: DrainDidNotComplete; Dependency: Kubernetes API; OrginalError: Drain did not complete pods [coredns-9d6c6c99b-twt4w]; AKSTeam: NodeProvisioning'. Check Pod Disruption Budgets; AKSTeam: NodeProvisioning
```

Reply:
The taint behavior only works in >= 1.10.3. However, there is a small bug that prevents the taints from being removed if they are added back, that's being fixed by https://github.com/cilium/cilium/pull/17112 and it will be part of 1.10.4.

Reply:
Thanks for the context. That clarifies it.